CodeLite/SocketAPI/clSocketBase.cpp

```diff
@@ -239,36 +271,28 @@ int clSocketBase::ReadMessage(wxString& message, int timeout)
     message_len = ::atoi(msglen);
 
     bytesRead = 0;
-    char* buff = new char[message_len + 1];
-    memset(buff, 0, message_len + 1);
+    std::unique_ptr<char> pBuff(new char[message_len]);

```

Comment:
I couldn't see the delete of buff in the success case so switched to a scoped array.